function/pkg/resources/unstructured/function_test.go

```diff
@@ -735,10 +763,13 @@ func Test_newGitFunction(t *testing.T) {
 								workspace.ResourceNameMemory: "10M",
 							},
 						},
-						"source":    "test-name",
-						"baseDir":   "test-base-dir",
-						"reference": "test-reference",
-						"type":      "git",
+						"source": map[string]interface{}{
+							"gitRepository": map[string]interface{}{
+								"url":       "",
```

Comment:
Why "url" is empty? Other values are filled in this test.

Reply:
Fixed